README.rst

```diff
@@ -112,35 +133,22 @@ In e.g. your ``init.lua``:
             -- Clip the infoview to a maximum width
             max_width = 79,
         },
-        -- Enable the Lean language server?
+        -- Enable the Lean3/Lean4 language servers?
         --
         -- false to disable, otherwise should be a table of options to pass to
         --  `leanls`. See https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#leanls
         -- for details though lean-language-server actually doesn't support all
         -- the options mentioned there yet.
         lsp = {
-            on_attach = function(client, bufnr)
-                -- See https://github.com/neovim/nvim-lspconfig#keybindings-and-completion
-                -- for detailed examples of what you may want to do here.
-                --
-                -- Mapping a key (typically K) to `vim.lsp.buf.hover()`
-                -- is highly recommended for Lean, since the hover LSP command
-                -- is where you'll see the current goal state.
-                --
-                -- You may furthermore want to add an `autocmd` to run it on
-                -- `CursorHoldI`, which will show the goal state any time the
-                -- cursor is unmoved in insert mode.
-                --
-                -- In the future, this plugin may offer a recommended "complete
-                -- setup" for easy enabling of the above.
-                local function buf_set_keymap(...) vim.api.nvim_buf_set_keymap(bufnr, ...) end
-                local function buf_set_option(...) vim.api.nvim_buf_set_option(bufnr, ...) end
-                buf_set_keymap('n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', {noremap = true})
-                buf_set_keymap('n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', {noremap = true})
-                buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')
-            end,
+            on_attach = on_attach,
             cmd = {"lean-language-server", "--stdio", '--', "-M", "4096"},
         }
+
+        -- replace the path below with a path to your Lean4 executable
+        lsp4 = {
+            on_attach = on_attach,
+            cmd = {"/path/to/lean4/executable/lean", "--server"},
```

Comment:
```suggestion
            cmd = {"lean", "--server"},
```
Can we make this the default?  `elan` should always pick the right version.  The vscode and emacs extensions also call `lean` by default.

Reply:
My only concern with this is that it may be misleading for simultaneous lean3/lean4 runs in particular. In this case, it seems that `lean-language-server` will attempt to run `lean --server` as well, which (unless `elan` is called internally to change the version in that environment) will be Lean 4.

Reply:
We could make this the default by setting `cmd_env.PATH` for lsp3 to include Lean 3 path and exclude Lean 4 path, but perhaps that's more trouble than it's worth.

Reply:
[Allowing the executable path to be given as an option](https://github.com/leanprover/lean-client-js/blob/f636b2572baf9579f12a2b90309ca81ac86bd44d/lean-language-server/src/index.ts#L30) could enable us to transfer this "full-path" requirement to lean3. Seems like the cleanest solution if we want this to be the default.

Reply:
I'm happy to merge a PR to lean-language-server that adds an option for the executable path.  If you want to override the PATH, another option would be to write a short shell script wrapper (like Julian does with his "Frankenlean").

> unless elan is called internally to change the version in that environment

99% of users use elan.  A handful use Nix, and then the executable is also called `lean` and in the PATH.  The plain `lean` command is thus a very good *default* setting, it will work in most situations out of the box.

(Maybe this wasn't clear because you haven't used `elan`.  Elan includes a `lean` wrapper.  So when you call `lean` it will automatically start the right version depending on the directory you're in.  If you're in a directory which uses Lean 3, `lean-language-server` would automatically start the right version of Lean 3.)

Reply:
I see - I was indeed not aware of the `lean` wrapper that `elan` introduces. That option for `lean-language-server` would be appreciated, but not at all a big deal at the moment - those of us who are too stubborn to move to an `elan`-compatible machine should be able to figure this out anyways!